docs/processes/qualityassurance/testing-and-dev.md

```diff
@@ -0,0 +1,199 @@
+
+When you are satisfied that your code is complete, it is wise to challenge your test again with known failing code to reaffirm your testing is still functioning as expected.
+
+**What next?**
+
+### TBC refactoring intro \*\*\*
+
+_Notes about things to expand/change below_
+
+```text
+Code quality:
+
+Is this reusable?
+
+Is this understandable?
+
+Is this readable?
+
+Can a third-party looking at your code easily understand what is going on?
+
+expand and entry into "refactoring code"
+```
+
+### **Refactoring Code**
+
+Refactoring involves changing the _internal_ structure to make it more easily understood, effective and/or cheaper to modify without changing the _external behaviour_ of the code.
+
+While writing a failing test case and then coding it to pass focuses on **requirements** (and adding **new functionality** ), refactoring focuses on **improving design**.
+
+**My code is complete and passing tests; is it time to refactor?**
+
+You don't need to assess your code for refactoring on every test, but if your find your code is:
+
+- repeating itself (or repeating functionality),
+- confusing to follow, or
+- you are getting stuck,
+
+these are good signs that it might be time to refactor.
+
+**How do I refactor?**
+
+Extract function refactoring is one common and intuitive approach to refactoring code: this involves comparing the code intention with the implementation. If the code is doing more than the intention, particularly if it is doing things that may be repeated elsewhere, it probably contains functions that should be split off.
+
+[Refactoring.com](http://refactoring.com/) provides useful reading that covers a number of refactoring techniques.
+
+**Further information about Test Driven Development**
+
+Deakin students have free access to LinkedIn Learning courses.
+
+A useful course on these general concepts is [Programming Foundations Test Driven Development](https://www.linkedin.com/learning/programming-foundations-test-driven-development-3/small-steps-to-great-things?u=2104084)
+
+Developers working with C++ may also find [Test Driven Development in C++](https://www.linkedin.com/learning/test-driven-development-in-c-plus-plus/welcome?autoplay=true&u=2104084) useful
+
+## Epics and User Stories
+
+# Epics
+
+Epics describe a large body of work that will be broken down into smaller tasks (called user stories). Epics are a higher-level view of user request or needs that help with considering larger aims, what value the work described will bring to the business, organizing work and creating a hierarchy. Epics will be broken down into smaller, more manageable tasks ("user stories") to be completed by team members. A Markdown Epic Template is also available.
+
+### TBC template link \*\*\*
+
+```block
```

Comment:
**issue:** Is `block` a valid language in GitHub markdown? It does not exist in the language [definition](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml).